cmd/lncli/commands.go

```diff
@@ -302,6 +302,117 @@ func sendCoins(ctx *cli.Context) error {
 	return nil
 }
 
+var bumpFeeCommand = cli.Command{
+	Name:      "bumpfee",
+	Category:  "On-chain",
+	Usage:     "Bumps the fee of an arbitrary input.",
+	ArgsUsage: "outpoint",
+	Description: `
+	Bump the fee of an arbitrary input within a transaction. This command
+	takes a different approach than bitcoind's bumpfee command. lnd has a
+	central batching engine in which inputs with similar fee rates are
+	batched together to save on transaction fees. Due to this, we cannot
+	rely on bumping the fee on a specific transaction, since transactions
+	can change at any point with the addition of new inputs. The list of
+	inputs that currently exist within lnd's central batching engine can be
+	retrieved through lncli pendingsweeps.
+
+	When bumping the fee of an input that currently exists within lnd's
+	central batching engine, a higher fee transaction will be created that
+	replaces the lower fee transaction through the Replace-By-Fee (RBF)
+	policy.
+
+	This command also serves useful when wanting to perform a
+	Child-Pays-For-Parent (CPFP), where the child transaction pays for its
+	parent's fee. This can be done by specifying an outpoint within the low
+	fee transaction that is under the control of the wallet.
+
+	If a fee preference isn't provided, either through the conf_target or
+	sat_per_byte parameters, then a fee rate corresponding to a confirmation
+	target of 6 blocks will be used.
+
+	Note that this command currently doesn't perform any validation checks
+	on the fee preference being provided. For now, the responsibility of
+	ensuring that the new fee preference is sufficient is delegated to the
+	user.`,
+	Flags: []cli.Flag{
+		cli.Uint64Flag{
+			Name: "conf_target",
+			Usage: "the number of blocks that the output should " +
+				"be swept on-chain within",
+		},
+		cli.Uint64Flag{
+			Name: "sat_per_byte",
+			Usage: "a manual fee expressed in sat/byte that " +
+				"should be used when sweeping the output",
+		},
+	},
+	Action: actionDecorator(bumpFee),
+}
+
+func bumpFee(ctx *cli.Context) error {
+	// Show command help if no arguments and flags were provided, or if more
+	// than the expected number of arguments were provided.
+	if ctx.NArg() == 0 && ctx.NumFlags() == 0 {
+		return cli.ShowCommandHelp(ctx, "bumpfee")
+	}
+	if ctx.NArg() > 1 && ctx.NumFlags() == 0 {
+		return cli.ShowCommandHelp(ctx, "bumpfee")
+	}
+
+	// Validate and parse the relevant arguments/flags.
+	var outpoint string
+	switch {
+	case ctx.Args().Present():
+		outpoint = ctx.Args().First()
+	default:
+		return fmt.Errorf("missing `outpoint` argument")
+	}
+
+	parts := strings.Split(outpoint, ":")
+	if len(parts) != 2 {
+		return errors.New("outpoint should be of the form txid:index")
+	}
+	txid := parts[0]
+	if hex.DecodedLen(len(txid)) != chainhash.HashSize {
+		return fmt.Errorf("invalid hex-encoded txid %v", txid)
+	}
+	outputIndex, err := strconv.Atoi(parts[1])
+	if err != nil {
+		return fmt.Errorf("invalid output index: %v", err)
+	}
```

Comment:
Extract to helper func (if not already present?)

Reply:
Can do so, but this parsing is somewhat specific to this command as it assumes that the outpoint will be the first argument provided.

Reply:
Yes, so I mean not the part that takes the `outpoint` string from the arg list, but converting that string into `lnrpc.OutPoint`

Reply:
Done.